frontend/src/app/featureFlags/FeatureFlagModal.tsx

```diff
@@ -68,71 +56,78 @@ const FeatureFlagModal: React.FC<Props> = ({
     </Grid>
   );
 
-  return (
-    <Modal data-testid="dev-feature-flags-modal" variant="large" isOpen onClose={onClose}>
-      <ModalHeader title="Feature Flags: Override Flags" />
-      <ModalBody>
-        <Tabs activeKey={activeTabKey} onSelect={(_event, tabIndex) => setActiveTabKey(tabIndex)}>
-          <Tab eventKey={0} title={<TabTitleText>Active</TabTitleText>} data-testid="activeFlagTab">
-            <Content style={{ height: '400px', padding: '8px' }}>
-              <Content component="p">
-                Feature flags default to the values defined in the dashboard config from the server.
-              </Content>
-              <Content component="p">
-                Flags that have an indeterminate value are because they are *not* defined in the
-                server.
-              </Content>
-              <h2> Tech Preview Flags </h2>
-              {renderFlags(Object.keys(techPreviewFlags), dashboardConfig)}
-              <h2> Temporary Developer Feature Flags</h2>
-              {renderFlags(Object.keys(devTemporaryFeatureFlags), dashboardConfig)}
-            </Content>
-          </Tab>
-          {devFlags.length > 0 && (
-            <Tab
-              eventKey={1}
-              title={<TabTitleText>Dev Flags</TabTitleText>}
-              data-testid="devFlagTab"
-            >
-              <Content style={{ height: '400px', padding: '8px' }}>
-                <Content component="p">
-                  Dev flags default to inactive and can only be changed for the current session.
-                </Content>
-                {renderFlags(devFlags)}
-              </Content>
-            </Tab>
-          )}
-          <Tab eventKey={2} title={<TabTitleText>Legacy</TabTitleText>} data-testid="legacyFlagTab">
-            <Content style={{ height: '400px', padding: '8px', overflow: 'scroll' }}>
-              <Content component="p">
-                These feature flags are hardly used and cannot be deleted yet at this time.
-              </Content>
-              <Content component="p">
-                Some of them may be deprecated soon; but most of them are permanent.
-              </Content>
-              <h2>Core Dashboard Flags</h2>
-              {renderFlags(Object.keys(coreDashboardFlags), dashboardConfig)}
-              <h2>Project & User Management Flags</h2>
-              {renderFlags(Object.keys(projectManagementFlags), dashboardConfig)}
-              <h2>Model Serving & AI/ML Infrastructure Flags</h2>
-              {renderFlags(Object.keys(modelServingFlags), dashboardConfig)}
-              <h2>Advanced AI/ML Features & Pipelines Flags</h2>
-              {renderFlags(Object.keys(advancedAIMLFlags), dashboardConfig)}
+  const buttonActions = [
+    {
+      label: 'Reset to defaults',
+      onClick: () => resetDevFeatureFlags(false),
+      variant: 'primary' as const,
```

Comment:
Could we type this as `ButtonProps['variant']` instead so call sites don’t need casts and we don’t block future variants?